src/windows/views/webview.py

```diff
@@ -329,40 +329,65 @@ def update_transition_data(self, transition_json, only_basic_props=True, ignore_
+        position = transition_data["position"]
+
+        # Determine if transition is intersecting a clip
+        # For example, the left side of a clip, or the right side, so we can determine
+        # which direction the wipe should be moving in
+        is_forward_direction = True
+        for intersecting_clip in Clip.filter(intersect=position):
+            log.debug(f"Intersecting Clip: {intersecting_clip}")
+            diff_from_start = abs(intersecting_clip.data.get("position", 0.0) - position)
+            diff_from_end = abs((intersecting_clip.data.get("position", 0.0) + \
+                                (intersecting_clip.data.get("end", 0.0) - intersecting_clip.data.get("start", 0.0))) \
+                                - position)
+            if diff_from_end < diff_from_start:
+                is_forward_direction = False
+        log.debug(f"Is transition moving a forward direction? {is_forward_direction}")
+
+        # Determine existing brightness and contrast ranges (if any)
+        brightness_range = []
+        contrast_range = []
+        if existing_item:
+            for point in existing_item.data["brightness"].get("Points", []):
+                point_value = float(point["co"]["Y"])
+                brightness_range.append(point_value)
+            for point in existing_item.data["contrast"].get("Points", []):
+                point_value = float(point["co"]["Y"])
+                contrast_range.append(point_value)
+        if not brightness_range:
+            brightness_range.extend([1, -1])
+        if not contrast_range:
+            contrast_range.extend([3])
+
+        # Create new brightness Keyframes (using the previous value range)
+        b = openshot.Keyframe()
+        if is_forward_direction:
+            b.AddPoint(1, sorted(brightness_range)[-1], openshot.BEZIER)
+            b.AddPoint(round(duration * fps_float), sorted(brightness_range)[0], openshot.BEZIER)
+        else:
+            b.AddPoint(1, sorted(brightness_range)[0], openshot.BEZIER)
+            b.AddPoint(round(duration * fps_float), sorted(brightness_range)[-1], openshot.BEZIER)
+        brightness = json.loads(b.Json())
+
+        # Create new contrast Keyframes (using the previous value range)
+        c = openshot.Keyframe()
+        if is_forward_direction:
+            c.AddPoint(1, sorted(contrast_range)[-1], openshot.BEZIER)
+            c.AddPoint(round(duration * fps_float), sorted(contrast_range)[0], openshot.BEZIER)
         else:
-            # Create new brightness and contrast Keyframes
-            b = openshot.Keyframe()
-            b.AddPoint(1, 1.0, openshot.BEZIER)
-            b.AddPoint(round(duration * fps_float) + 1, -1.0, openshot.BEZIER)
-            brightness = json.loads(b.Json())
+            c.AddPoint(1, sorted(contrast_range)[0], openshot.BEZIER)
+            c.AddPoint(round(duration * fps_float), sorted(contrast_range)[-1], openshot.BEZIER)
+        contrast = json.loads(c.Json())
```

Comment:
So rather than overwriting contrast and brightness keyframes, we follow along, and then continue the fade from the first keyframe to the first frame, or from the last keyframe to the last frame?

Makes sense as a way to do it, just making sure that's the intent here.

Reply:
contrast_range contains the actual values used on the Y axis for brightness - for example, imagine a user set the following values [3.0, -3.0].  Those would be sorted first, and our new brightness keyframe would either be [3.0, -3.0] if it's on the left side of a clip, or [-3.0, 3.0] if it's on the right of a clip. The idea though, is to preserve the user's actual values if the keyframe already exists, and recreate a new keyframe (going in the correct wipe direction).

Reply:
Most average users do not customize the `brightness` and `contrast` keyframes used by transitions. So, this is kind of a quick and dirty way to preserve the customized values (if any) when resizing transitions / moving transitions. This isn't perfect, because we are only preserving the min/max range of values used in the keyframe... not actually all the points in between (if there are more than 2).

Reply:
Ok.... So if I'm reading this right, then if there's only 1 brightness keyframe, then there will be no change in brightness.

Because `brightness_range[0]` and `brightness_range[-1]` will be the same value.

Reply:
Exactly... it works with a range of 1 to n